src/pathfinding_service/middleware.py

```diff
@@ -1,25 +1,26 @@
-from typing import Tuple
+from typing import Any, Callable, Tuple
 
 import gevent
 from requests.exceptions import ConnectionError, HTTPError, Timeout, TooManyRedirects
+from web3 import Web3
 from web3.middleware.exception_retry_request import check_if_retry_on_failure
 
 
 def http_retry_with_backoff_middleware(
-    make_request,
-    web3,
+    make_request: Callable,
+    web3: Web3,
     errors: Tuple = (ConnectionError, HTTPError, Timeout, TooManyRedirects),
     retries: int = 10,
     first_backoff: float = 0.2,
     backoff_factor: float = 2,
-):
+) -> Callable:
     """ Retry requests with exponential backoff
 
     Creates middleware that retries failed HTTP requests and exponentially
     increases the backoff between retries. Meant to replace the default
     middleware `http_retry_request_middleware` for HTTPProvider.
     """
-    def middleware(method, params):
+    def middleware(method: str, params: dict) -> Any:
```

Comment:
I prefer the capitalized version `Dict` for type annotations, but have no strong objections.

Reply:
Since there are no uppercase variants of `str`, `int`, etc we will have a mixture of upper and lowercase, anyway. So I don't see and advantage in always using the uppercase when available. Using the lowercase saves adding and removing a few imports on code changes. No strong opinions here either, but I'm slightly on the side of using lowercase when the additional features of the uppercase variants are not needed. I couldn't find any official recommendations for either.

Reply:
👍